test/org/sagebionetworks/bridge/services/StormPathUserAdminServiceTest.java

```diff
@@ -150,4 +163,31 @@ public void testDeleteUserThatHasBeenDeleted() {
         userAdminService.deleteUser(study, session.getUser().getEmail());
         assertNull(authService.getSession(session.getSessionToken()));
     }
+    
+    @Test
+    public void creatingUserConsentsToAllRequiredConsents() {
+        Study study = null;
+        try {
+            study = TestUtils.getValidStudy(StormPathUserAdminServiceTest.class);
+            studyService.createStudy(study);
+            
+            // Create another required subpopulation
+            Subpopulation subpop = Subpopulation.create();
+            subpop.setName("Another subpopulation");
+            subpop.setRequired(true);
+            subpopService.createSubpopulation(study, subpop);
+            
+            UserSession session = userAdminService.createUser(signUp, study, null, true, true);
+            List<ConsentStatus> statuses = new ArrayList<ConsentStatus>(session.getUser().getConsentStatuses().values());
+            
+            // There are two consents, the user is consented to both of them, the session reflects this
+            assertTrue(statuses.get(0).isConsented());
+            assertTrue(statuses.get(1).isConsented());
```

Comment:
I just moved this test to use mocks. The problem is that the mock consentService isn't going to update the consent status records, nor should it in the test. So all this test will test is that consentService.consentToResearch() is called for all the consents (or that if a sub pop guid is provided, that is used instead).